astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2578,3 +2585,170 @@ def test_all_included(self):
+                    ), f"argument {nt!r} is not re-exposed as keyword"
+                elif kt is POSITIONAL_OR_KEYWORD:
+                    assert (
+                        have_args_helper and have_kwargs_helper
+                    ), f"argument {nt!r} is not re-exposed as positional-or-keyword"
+            if pt.kind is VAR_POSITIONAL:
+                assert have_args_helper, "helper is missing a catch-all *args argument"
+            if pt.kind is VAR_KEYWORD:
+                assert (
+                    have_kwargs_helper
+                ), "helper is missing a catch-all **kwargs argument"
+
+            pos_target += 1
+
+    def test_known_arguments(self, functions):
+        # validate that all exposed arguments map to something in the target
+        target, helper = functions
+
+        try:
+            sig_target = inspect.signature(target)
+        except ValueError:
+            pytest.skip("Non Python function cannot be inspected at runtime")
+
+        params_target = sig_target.parameters
+        sig_helper = inspect.signature(helper)
+        params_helper = sig_helper.parameters
+
+        # a : positional-only
+        positionals_only_target = self._get_param_group(
+            params_helper, [POSITIONAL_ONLY]
+        )
+        positionals_only_helper = self._get_param_group(
+            params_helper, [POSITIONAL_ONLY]
+        )
+
+        if (nhelper := len(positionals_only_helper)) > (
+            ntarget := len(positionals_only_target)
+        ):
+            unknown: list[str] = positionals_only_helper[ntarget:]
+            raise AssertionError(
+                "Found unknown positional-only parameter(s) in helper's signature: "
+                f"{unknown}, at position(s) {list(range(ntarget, nhelper))}"
+            )
+
+        # b: positional-or-keyword
+        # TODO: reduce code duplication with pytest.mark.parametrize ?
+        positional_or_keyword_target = self._get_param_group(
+            params_target, [POSITIONAL_OR_KEYWORD]
+        )
+        positional_or_keyword_helper = self._get_param_group(
+            params_helper, [POSITIONAL_OR_KEYWORD]
+        )
+
+        if (nhelper := len(positional_or_keyword_helper)) > (
+            ntarget := len(positional_or_keyword_target)
+        ):
+            unknown: list[str] = positional_or_keyword_helper[ntarget:]
+            raise AssertionError(
+                "Found unknown positional-or-keyword parameter(s) in helper's signature: "
+                f"{unknown}, at position(s) {list(range(ntarget, nhelper))}"
+            )
+
+        # c: keyword-allowed
+        keyword_allowed_target = set(
+            self._get_param_group(params_target, [KEYWORD_ONLY, POSITIONAL_OR_KEYWORD])
+        )
+        keyword_allowed_helper = set(
+            self._get_param_group(params_helper, [KEYWORD_ONLY, POSITIONAL_OR_KEYWORD])
+        )
+
+        # additional private keyword-only argument are allowed because
```

Comment:
Do we need this?

Reply:
yes, I added it after I discovered that this pattern was used in `def quantile(a, q, *args, _q_unit=dimensionless_unscaled, **kwargs):`

Reply:
Ah, yes. Maybe refer to it in the comment?